File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> triangle) {
    int answer = 0;
    int h = triangle.size();
    for (int i = 1; i < h; i++) {
        triangle[i][0] += triangle[i-1][0];
        for (int j = 1; j < i; j++) {
            triangle[i][j] += max(triangle[i-1][j-1], triangle[i-1][j]);
        }
        triangle[i][i] += triangle[i-1][i-1];
    }      
    for (int i = 0; i < h; i++) {
        answer = max(answer, triangle[h-1][i]);
    }
    return answer;
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    answer = 0;
    
    height = len(triangle)  # 세로 크기
    
    i = 1
    for i in range(1, height):
        t = len(triangle[i])
        for j in range(t):  
            if j == 0:  # 왼쪽 모서리 인덱스인 경우
                triangle[i][j] += triangle[i-1][j]
            elif j == t - 1:  # 오른쪽 모서리 인덱스인 경우
                triangle[i][j] += triangle[i-1][j-1]
            else:
                triangle[i][j] += max(triangle[i-1][j-1], triangle[i-1][j])
                
        answer = max(triangle[height-1])
    return answer